### ISSUE TITLE
MA0028 does not report last documented example

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#1030]
### Version of the Meziantou.Analyzer NuGet package

3.0.10

### Rule Identifier

MA0028

### Target Framework

.NET 10.0

### C# Language version

C# 14

### Description

MA0028 is not reported for the last documented example [here](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0028.md).

### Reproduction Steps

```c#
new StringBuilder().Append(10.ToString("n1", CultureInfo.CurrentCulture));

// Should be
new StringBuilder().AppendFormat(CultureInfo.CurrentCulture, "{0:n1}", 10);
```

**Comment by meziantou** [comment]
The documentation needs to be updated. The ToString version is faster.

| Method               | Mean     | Error    | StdDev    | Ratio | RatioSD | Gen0   | Allocated | Alloc Ratio |
|--------------------- |---------:|---------:|----------:|------:|--------:|-------:|----------:|------------:|
| Case1_AppendToString | 44.99 ns | 2.172 ns |  6.371 ns |  1.02 |    0.21 | 0.0081 |     136 B |        1.00 |
| Case2_AppendFormat   | 70.19 ns | 4.112 ns | 11.994 ns |  1.59 |    0.36 | 0.0076 |     128 B |        0.94 |
